src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,7 +60,18 @@ public boolean equals(Object other) {
      * Two names are considered similar if ...
      */
     public boolean isSimilar(Name other) {
-        return this.equals(other);
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof Name)) {
```

Comment:
this cannot happen because in method signature you've indicated `other` to be `Name` type.